build/gated_jobs/build_and_run_tests.yml

```diff
@@ -0,0 +1,109 @@
+# This is the job template for building the binaries for one configuration. It does:
+# - setup the machine
+# - build binaries for desired configuration
+# - upload binaries artifacts for the desired configuration
+
+parameters:
+  - name: BUILD_SUFFIX
+    type: string
+  - name: GBALLOC_LL_TYPE
+    type: string
+  - name: FSANITIZE_TYPE
+    type: string
+  - name: build_configuration
+    default: "Debug"
+  - name: ARCH_TYPE
+    default: "x64"
+
+jobs:
+- job: build_and_run_tests_${{ parameters.BUILD_SUFFIX }}
+  displayName: 'Build and run tests, build_configuration: ${{ parameters.build_configuration }}, ${{ parameters.GBALLOC_LL_TYPE}}, ARCH_TYPE=${{ parameters.ARCH_TYPE }}, FSANITIZE_TYPE: ${{ parameters.FSANITIZE_TYPE}}'
+
+  variables:
+    ${{ if eq(parameters.FSANITIZE_TYPE, 'ON') }}:
+      use_vld_option: '-Duse_vld:bool=OFF'
+    ${{ if eq(parameters.FSANITIZE_TYPE, 'OFF') }}:
+      use_vld_option: '-Duse_vld:bool=ON'
+
+  pool:
+    name: Azure-MessagingStore-WinBuildPoolVS2022
+    demands:
+    - Cmd
+    - msbuild
+    - cmake
+    - visualstudio
+
+  workspace:
+    clean: all
+
+  steps:
+  - checkout: self
+    submodules: true
+    clean: false
+    persistCredentials: true
```

Comment:
>persistCredentials [](http://example.com/codeflow?start=4&length=18)

just wondering why this is needed.

Reply:
I believe that is a copy/pasta remnant from the main repo. Not needed => removed.